cmd/clusterctl/client/alpha/rollout_restarter_test.go

```diff
@@ -35,13 +36,15 @@ func Test_ObjectRestarter(t *testing.T) {
 		ref  corev1.ObjectReference
 	}
 	tests := []struct {
-		name           string
-		fields         fields
-		wantErr        bool
-		wantAnnotation bool
+		name        string
+		apiType     string
```

Comment:
Can't we just use the obj for determining the type and drop the apiType field (same for other occurencies in tests)? E.g.:

```
switch v := obj.(type) {
case clusterv1.MachineDeployment:
	...
case controlplanve1.KubeadmControlPlane:
	...
default:
	// some error handling/ginkgo except
}
```

Reply:
Really nice idea, thanks! 🙂 👍🏻 ➕